app/frontends/stac.ts

```diff
@@ -81,11 +122,19 @@ export async function getStacItem(req, res): Promise<void> {
   const { jobId, itemIndex } = req.params;
   const keys = keysToLowerCase(req.query);
   const linkType = keys.linktype?.toLowerCase();
+  const itemIndexInt = parseInt(itemIndex, 10);
   try {
+    if (itemIndexInt === undefined) {
+      throw new RequestValidationError('STAC item index should be a valid integer');
+    }
+    const pagingParams: PagingParams = { page: itemIndexInt + 1, limit: 1 };
     await handleStacRequest(
       req,
       res,
-      (data) => stacItemCreate.apply(null, [data, itemIndex, linkType]),
+      (job: Job) => stacItemCreate.apply(
+        null, [job.jobID, job.request, job.links[0], itemIndexInt, linkType, job.createdAt],
```

Comment:
handleStackRequest, which calls this callback method `(job: Job) => stacItemCreate.apply(...)`: 

1. validates that the STAC links requested by the user actually exist before invoking the callback 
2. loads only the requested links, (one of them in this case), into job.links via Job.byUsernameAndRequestId()
 
Assuming handleStackRequest doesn't change too much, you can assume that the requested STAC link does exist, and only one was loaded into job.links (since that's all that we need).